### ISSUE TITLE
Possible feature: use system cmake if found and meets required minimum version

### ISSUE THREAD

**Issue by tkelman** [JuliaPackaging/CMakeWrapper.jl#4]
Could have some way of packages specifying what cmake version they need, and if this package finds a good enough cmake on the path it would use it instead of needing to download a private copy every time? It's not a huge download, but it does seem a bit wasteful if there's already a copy installed, especially on architectures where this package would otherwise need to build cmake from source.

**Comment by rdeits** [comment]
Yeah, I would like to do that eventually. I chose the current design because I'm building a couple of projects that require ridiculously modern cmake versions (like 3.5-3.7), so system cmake is almost never good enough. But yes, it would be nice to defer to the system cmake if it's good enough for a particular build.

**Comment by tkelman** [comment]
Has cmake really been adding features recently that people are already wanting to use? I thought LLVM was pretty arbitrary in picking 3.4.3 and that's the newest requirement I've encountered.

**Comment by rdeits** [comment]
The one I deal with most often is our lab's code in https://github.com/RobotLocomotion/drake which requires cmake 3.5 and uses features back-ported from 3.7 via some extra modules. Some of its dependencies have also been upgraded to require at least 3.5. I don't know what features are actually being used, though (we have some external developers who are handling the build systems).
